internal/p2p/peermanager.go

```diff
@@ -527,6 +594,18 @@ func (m *PeerManager) DialFailed(ctx context.Context, address NodeAddress) error
 
 	addressInfo.LastDialFailure = time.Now().UTC()
 	addressInfo.DialFailures++
+
+	// If a dial fails more than MaxFailedDialAttempts we should
+	// mark it inactive and not attempt to dial it again.
+	var totalDialFailures uint32
+	for _, addr := range peer.AddressInfo {
+		totalDialFailures += addr.DialFailures
+	}
+	if m.options.MaxFailedDialAttempts > 0 && totalDialFailures > m.options.MaxFailedDialAttempts {
+		peer.Inactive = true
+		m.metrics.PeersInactivated.Add(1)
+	}
+
```

Comment:
I added this in response to Callum's feedback, I'm sort of indifferent on the topic and could go either way, and think there are merits to both. 

I think it could go either way: I think we should avoid just spinning on dialing peers we can't connect to and the current maxFailedDialAttempts value (1024) is pretty high: the policy once inactivated is "call us when you're ready," and the split of tracking of incoming/outgoing connections makes it more likely that there are going to be slots for peers to make incoming connections.

Either peers we can't dial have gone away entirely (so inactivated is good,) or they're offline for a while and will dial us eventually (fine), but if we spin on trying to dial a peer that's bad it means we won't end up dialing a *different* peer that might be much better.

Reply:
I've removed the inactivation after dialing failures code.